packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -48,20 +52,34 @@ export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
             fileName={truncatedFileName(file.name, truncateLength)}
             key={file.name}
             fileExtension={extension(file.name)}
-            actionIcon={<DownloadIconTrampoline />}
+            actionIcon={
+              showSpinner ? (
+                <Spinner size={SpinnerSize.medium} aria-live={'assertive'} />
+              ) : (
+                <ChatCompositeIcon iconName="Download" />
+              )
+            }
             actionHandler={() => {
-              props.downloadHandler && props.downloadHandler(userId, file);
+              if (props.downloadHandler) {
+                setShowSpinner(true);
+                props
+                  .downloadHandler(userId, file)
+                  .then((value: URL | FileDownloadErrorMessage) => {
+                    if (value instanceof URL) {
+                      window.open(value.toString(), '_blank', 'noopener,noreferrer');
+                    } else {
+                      //TODO: implement error handling for reject
+                      console.log(value);
```

Comment:
nit: remove the console log (I understand its temporary until the todo is done but for PRs they should still be removed -- typically because todo's can sometimes take a long time/forgot to be done and users my have weird messages in the console)

Reply:
done